streamalert_cli/outputs/handler.py

```diff
@@ -78,25 +180,271 @@ def handler(cls, options, config):
-        if output_exists(output_config, props, service):
+        if not options.update and output_exists(config['outputs'], props, service):
+            # If the output already exists and update is not set
+            # ask for user input again for a unique configuration
             return cls.handler(options, config)
 
-        provider = OutputCredentialsProvider(service, config=config, region=region, prefix=prefix)
-        result = provider.save_credentials(props['descriptor'].value, kms_key_alias, props)
-        if not result:
-            LOGGER.error('An error occurred while saving \'%s\' '
-                         'output configuration for service \'%s\'', props['descriptor'].value,
-                         options.service)
+        if not cls.save_credentials(service, config, props):
+            # Error message is already logged so no need to log a new one
             return False
 
-        updated_config = output.format_output_config(output_config, props)
-        output_config[service] = updated_config
-        config.write()
+        cls.update_config(options, config, props, output, service)
 
         LOGGER.info('Successfully saved \'%s\' output configuration for service \'%s\'',
-                    props['descriptor'].value, options.service)
+                    props['descriptor'].value, service)
         return True
+
+
+class OutputSetFromFileSubCommand(CLICommand, OutputSharedMethods):
+    description = 'Set numerous outputs from a file'
+
+    @classmethod
+    def setup_subparser(cls, subparser):
+        """Setup: manage.py output set-from-file [options]
+
+        Args:
+            outputs (list): List of available output services
+        """
+        set_from_file_parser = generate_subparser(
+            subparser,
+            'set-from-file',
+            description=cls.description,
+            help=cls.description,
+            subcommand=True
+        )
+
+        # Add the optional file flag
+        set_from_file_parser.add_argument(
+            '--file',
+            '-f',
+            default=OUTPUTS_FILE,
+            help='Path to the json file, relative to the current working directory'
+        )
+
+        # Add the optional update flag, which allows existing outputs to be updated
+        set_from_file_parser.add_argument(
+            '--update',
+            '-u',
+            action='store_true',
+            default=False,
+            help='Allow existing outputs to be overwritten'
+        )
+
+    @classmethod
+    def handler(cls, options, config):
+        """Configure multiple outputs for multiple services
+        Args:
+            options (argparse.Namespace): Basically a namedtuple with the service setting
+            config (StreamAlert.config): The configuration of StreamAlert
+        Returns:
+            bool: False if errors occurred, True otherwise
+        """
+        with open(options.file, "r") as json_file_fp:
+            file_contents = json.load(json_file_fp)
```

Comment:
Probably we should do exception handling here because the file may have syntax error or not a json file. Exit if the file is not a json format with error message.

Reply:
This would make a useful helper method that's for sure!